concent_api/core/tests/test_integration_accept_or_reject_results.py

```diff
@@ -0,0 +1,1082 @@
+                subtask_id              = '2',
+                payment_ts              = "2018-02-05 11:00:01",
+            )
+        )
+
+        with mock.patch('core.views.is_provider_account_status_positive', get_provider_account_status_true_mock):
+            with freeze_time("2018-02-05 11:00:01"):
+                response_1 = self.client.post(
+                    reverse('core:send'),
+                    data                            = serialized_force_subtask_results_response,
+                    content_type                    = 'application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY  = self._get_encoded_requestor_public_key(),
+                )
+
+        self._test_400_response(response_1)
+
+    def test_requestor_sends_subtask_results_rejection_but_provider_does_not_submitted_force_subtask_results_concent_should_reject_it(self):
+        """
+        Test if Requestor want submit ForceSubtaskResultsResponse with SubtaskResultsRejected message,
+        but Provider doesn't submitted ForceResultsAccepted before
+
+
+        Exptected message exchange:
+        Requestor   -> Concent:     ForceSubtaskResultsResponse with SubtaskResultsRejected
+        Concent     -> Requestor:   HTTP 400
+        """
+        serialized_force_subtask_results_response = self._get_serialized_force_subtask_results_response(
+            timestamp               = "2018-02-05 11:00:00",
+            requestor_private_key   = self.REQUESTOR_PRIVATE_KEY,
+            subtask_results_rejected = self._get_deserialized_subtask_results_rejected(
+                timestamp               = "2018-02-05 11:00:00",
+                reason                  = message.tasks.SubtaskResultsRejected.REASON.VerificationNegative,
+                report_computed_task    = self._get_deserialized_report_computed_task(
+                    timestamp   = "2018-02-05 11:00:00",
+                    subtask_id  = 'xxyyzz',
+                    task_to_compute = self._get_deserialized_task_to_compute(
+                        timestamp   = "2018-02-05 11:00:00",
+                        deadline    = "2018-02-05 11:00:05",
+                        task_id     = 'xxyyzz',
+                    )
+                )
+            )
+        )
+
+        with mock.patch('core.views.is_provider_account_status_positive', get_provider_account_status_true_mock):
+            with freeze_time("2018-02-05 11:00:01"):
+                response_1 = self.client.post(
+                    reverse('core:send'),
+                    data                            = serialized_force_subtask_results_response,
+                    content_type                    = 'application/octet-stream',
+                    HTTP_CONCENT_CLIENT_PUBLIC_KEY  = self._get_encoded_requestor_public_key(),
+                )
+
+        self._test_400_response(response_1)
+
+    def test_requestor_sends_subtask_results_accepted_and_concent_should_return_it_to_provider(self):
```

Comment:
This test is missing step when Requestor received `ForceSubtaskResults`. But actually it discovers bug or an issue - should it be possible to send `SubtaskResultsAccepted` by requestor before receiving `ForceSubtaskResults` ? I think not.

Reply:
Actually I haven't known that message should be delivered but when Concent will be working with Golem it will be normal to deliver first this message, becuase Requestor won't know which task he will want to accept or reject. But yes, it is bug and we need issue for that